src/Servers/Kestrel/Core/src/Internal/Http/IHttpRequestLineHandler.cs

```diff
@@ -24,13 +24,13 @@ public HttpVersionAndMethod(HttpMethod method, int methodEnd)
 
     public HttpVersion Version
     {
-        get => (HttpVersion)(sbyte)(byte)_versionAndMethod;
+        readonly get => (HttpVersion)(sbyte)(byte)_versionAndMethod;
         set => _versionAndMethod = (_versionAndMethod & ~0xFFul) | (byte)value;
     }
 
-    public HttpMethod Method => (HttpMethod)(byte)(_versionAndMethod >> 8);
+    public readonly HttpMethod Method => (HttpMethod)(byte)(_versionAndMethod >> 8);
 
-    public int MethodEnd => (int)(uint)(_versionAndMethod >> 32);
+    public readonly int MethodEnd => (int)(uint)(_versionAndMethod >> 32);
```

Comment:
Is this a public type is in an internal namespace? There is some discussion in the runtime about how changing a readonly member to non-readonly in a later version would be considered a breaking change: https://github.com/dotnet/runtime/issues/1718#issuecomment-479706525, something to be aware of.

Reply:
TIL it is a breaking change to remove readonly from structs.

I don't see a scenario when these properties would ever not be readonly so I think this is fine.